Userland/Libraries/LibCore/Process.cpp

```diff
@@ -160,9 +162,37 @@ ErrorOr<bool> Process::is_being_debugged()
         return (tracer_pid != 0UL);
     }
     return false;
+#elif defined(AK_OS_MACOS)
```

Comment:
Not sure if it's true or not, but is this sysctl common b/w all the popular BSDs? It's possible we could use another AK_OS_BSD_GENERIC here.

Reply:
The sysctl is common but as far as I can tell the used structs are not identical:

https://github.com/search?q=repo%3Aopenbsd%2Fsrc%20P_TRACED&type=code

Reply:
Pinging @nipos, do you know about this?

Reply:
Quickly looking at https://github.com/search?q=repo%3Aopenbsd%2Fsrc%20P_TRACED&type=code ,I can find some incompatible differences.
Simply making this AK_OS_BSD_GENERIC won't work here because FreeBSD uses ki_flag instead of p_flag which both NetBSD and MacOS are using and NetBSD calls its kinfo_proc stuct "struct ::kinfo_proc2".
For OpenBSD and Solaris,things may again be very different but I don't have a clear answer about them for now.
I recommend limiting this feature to MacOS at first.
Extending it at least to FreeBSD and NetBSD seems like a trivial task that I can do when I have some more time again.